### PR TITLE
config: use pflag to let viper override cmd flags

### LINKED ISSUE #1094: Enable multithreaded Config not respecting toml
I have  following `dice.toml` config:

<details>
<summary>dice.toml</summary>

```toml
Version = '0.0.4'
InstanceID = ''

[AsyncServer]
Addr = '0.0.0.0'
Port = 7379
KeepAlive = 300
Timeout = 300
MaxConn = 0

[HTTP]
Enabled = true
Port = 8082

[WebSocket]
Enabled = true
Port = 8379
MaxWriteResponseRetries = 3
WriteResponseTimeout = 10000000000

[Performance]
WatchChanBufSize = 20000
ShardCronFrequency = 1000000000
MultiplexerPollTimeout = 100000000
MaxClients = 20000
EnableMultiThreading = true
StoreMapInitSize = 1024000
AdhocReqChanBufSize = 20

[Memory]
MaxMemory = 0
EvictionPolicy = 'allkeys-lfu'
EvictionRatio = 0.9
KeysLimit = 200000000
LFULogFactor = 10

[Persistence]
AOFFile = './dice-master.aof'
PersistenceEnabled = true
WriteAOFOnCleanup = false

[Logging]
LogLevel = 'debug'
PrettyPrintLogs = true

[Auth]
UserName = 'dice'
Password = ''

[Network]
IOBufferLength = 512
IOBufferLengthMAX = 51200
```
</details>

and I am starting the server as: `go run main.go -c dice.toml`.

### Expected behaviour
```
2024/10/14 23:05:33 INFO configurations loaded successfully.
11:05PM DBG The DiceDB server has started in multi-threaded mode. number of cores=8
11:05PM INF RESP Server successfully bound Host=0.0.0.0 Port=7379
11:05PM INF DiceDB ready to accept connections on port resp-port=7379
11:05PM INF Websocket Server running port=8379
```

### Actual behaviour
```
2024/10/14 23:12:23 INFO configurations loaded successfully.
11:12PM DBG The DiceDB server has started in single-threaded mode.
11:12PM INF DiceDB server is running in a single-threaded mode port=7379 version=0.0.4
11:12PM WRN DiceDB is running without authentication. Consider setting a password.
11:12PM INF HTTP Server running addr=:8082
11:12PM INF Websocket Server running port=8379
```

The `EnableMultiThreading = true` isn't having any effect on the dice startup

### PR DESCRIPTION
**Pull request by codeasashu**
This fixes #1094

### CODE REVIEW

**Comment by codeasashu** [comment]
@JyotinderSingh Take a look at this one too, its quite small one to be merged

**Comment by lucifercr07** [comment]
I'm not sure if we'd be investing much time with viper related changes as new config is in work.
@JyotinderSingh/@vinitparekh17 thoughts?

**Comment by vinitparekh17** [comment]
@lucifercr07 Since we agreed to eliminate most of these flags and simplify our implementation to be primarily config file-based, except for a few sub-commands like `-help` and `-version`, merging this PR may not make sense at this point. However, until we roll out those discussed changes, you may proceed with merging this PR to acknowledge the author’s efforts.

**Review by lucifercr07 — APPROVED** [review]

File: config/config.go [hunk 1]
```diff
@@ -222,11 +224,38 @@ var baseConfig = Config{
 
 var defaultConfig Config
 
+func setupFlags() {
```
Comment (lucifercr07) [inline config/config.go:227]: Lot of these values have been changed with recent commits. @codeasashu can you please resolve accordingly.
Rest LGTM.

**Review by lucifercr07 — CHANGES_REQUESTED** [review]

File: config/config.go (hunk 1 above)
Comment (codeasashu) [reply config/config.go:227]: Sure, will do shortly

Comment (codeasashu) [reply config/config.go:227]: rebased

**Comment by JyotinderSingh** [comment]
@vinitparekh17 is this PR still relevant based on the new changes?

**Comment by vinitparekh17** [comment]
> @vinitparekh17 is this PR still relevant based on the new changes?

No it is not.

**Pull request closed without merging**